### ISSUE TITLE
jazzy crashes during site building with "in `+': no implicit conversion of nil into String

### ISSUE THREAD

**Issue by juangamnik** [realm/jazzy#171]
After some time of not regenerating the jazzy files for my project I tried it again today and got an error. So I updated to the newest version 0.1.2 and tried again. Still the same error:

```bash
building site
/Library/Ruby/Gems/2.0.0/gems/jazzy-0.1.2/lib/jazzy/sourcekitten.rb:41:in `+': no implicit conversion of nil into String (TypeError)
	from /Library/Ruby/Gems/2.0.0/gems/jazzy-0.1.2/lib/jazzy/sourcekitten.rb:41:in `block in make_doc_urls'
	from /Library/Ruby/Gems/2.0.0/gems/jazzy-0.1.2/lib/jazzy/sourcekitten.rb:32:in `each'
	from /Library/Ruby/Gems/2.0.0/gems/jazzy-0.1.2/lib/jazzy/sourcekitten.rb:32:in `make_doc_urls'
	from /Library/Ruby/Gems/2.0.0/gems/jazzy-0.1.2/lib/jazzy/sourcekitten.rb:37:in `block in make_doc_urls'
	from /Library/Ruby/Gems/2.0.0/gems/jazzy-0.1.2/lib/jazzy/sourcekitten.rb:32:in `each'
	from /Library/Ruby/Gems/2.0.0/gems/jazzy-0.1.2/lib/jazzy/sourcekitten.rb:32:in `make_doc_urls'
	from /Library/Ruby/Gems/2.0.0/gems/jazzy-0.1.2/lib/jazzy/sourcekitten.rb:37:in `block in make_doc_urls'
	from /Library/Ruby/Gems/2.0.0/gems/jazzy-0.1.2/lib/jazzy/sourcekitten.rb:32:in `each'
	from /Library/Ruby/Gems/2.0.0/gems/jazzy-0.1.2/lib/jazzy/sourcekitten.rb:32:in `make_doc_urls'
	from /Library/Ruby/Gems/2.0.0/gems/jazzy-0.1.2/lib/jazzy/sourcekitten.rb:194:in `parse'
	from /Library/Ruby/Gems/2.0.0/gems/jazzy-0.1.2/lib/jazzy/doc_builder.rb:108:in `build_docs_for_sourcekitten_output'
	from /Library/Ruby/Gems/2.0.0/gems/jazzy-0.1.2/lib/jazzy/doc_builder.rb:67:in `build'
	from /Library/Ruby/Gems/2.0.0/gems/jazzy-0.1.2/bin/jazzy:15:in `<top (required)>'
	from /usr/local/bin/jazzy:24:in `load'
	from /usr/local/bin/jazzy:24:in `<main>'
```

I have installed Xcode Version 6.1.1 (6A2006) and Version 6.3 (6D520o). I am using the beta version of 6.3 and the `xcode-select` path is set to `/Applications/Xcode-beta.app/Contents/Developer`. In #124 @jpsim added an output to STDERR, which swift file couldn't be parsed by sourcekitten, which helped a lot to find the problem. The comments on the issue #124 tell, that this feature has improved (not only STDERR, but direct output from the JSON parser). But unfortunately it does not output, which file is problematic. I think, because this time it is not a JSON problem but some kind of missing information that now leads to a null pointer. If I knew, which of my files is the nasty guy...

**Comment by segiddins** [comment]
What command did you use to run jazzy?

**Comment by juangamnik** [comment]
`jazzy`

**Comment by jpsim** [comment]
@juangamnik running SourceKitten directly on this project should show us where it's failing. If you send me the Xcode project that SK is failing to parse, I'll be able to troubleshoot and fix faster.

**Comment by segiddins** [comment]
@jpsim it's a `nil` USR

**Comment by jpsim** [comment]
@segiddins sure, but on what? I'd like to know what type I'm incorrectly assuming to always have a USR.

**Comment by juangamnik** [comment]
@jpsim its a closed/private project I cannot send it. But I can execute some command on it and send you the result.

Anyways, IMO jazzy/sourcekitten should -- at any time you get aware of an error -- be enhanced with a nice error message. Otherwise it will become a support hell.

**Comment by jpsim** [comment]
> Anyways, IMO jazzy/sourcekitten should -- at any time you get aware of an error -- be enhanced with a nice error message. Otherwise it will become a support hell.

Of course! If there was a silver bullet to never crash unexpectedly and log an error message instead, we'd do it :wink:. When we find the root of this current issue, we'll be sure to a) fix it and b) add logging in the failure case. But I guarantee you, there will always be another bug `¯\_(ツ)_/¯`.

**Comment by juangamnik** [comment]
> Of course! If there was a silver bullet to never crash unexpectedly and log an error message instead, we'd do it :wink:. When we find the root of this current issue, we'll be sure to a) fix it and b) add logging in the failure case. But I guarantee you, there will always be another bug.

That's for sure and I'm Award, nur still, there are some who discourage error message for whatever reason or like it cryptic/generic, but in general this are not the users (git itself is a bad example for that in many cases). But if this is already your policy I didn't say anything. It's just that last time there was a JSON prob and now there is a error log which file couldn't be parsed. Now there is somewhere in another (later?) stage a problem. Why does jazzy not tell the problematic file. This is an information you should have always (as all your info seems to be related to either a file, a command line argument, access to the file say stem, or a missing dependency) and therefore be able to log it in an error case. Please understand this as a constructive suggestion. :wink:

**Comment by juangamnik** [comment]
Written via smartphone (sorry for the autocorrection)

**Comment by maplesteve** [comment]
Hi!

Having the same problem here.
I ran `/Library/Ruby/Gems/2.0.0/gems/jazzy-0.1.2/lib/jazzy/sourcekitten/sourcekitten doc` on the project and compared the output (it doesn't crash) with the `Bicycle.json` (fixture from the sourcekitten repo). The JSON doesn't have any entries for either
- `key.parsed_declaration`
- `key.usr`
- `key.doc.full_as_xml`
- `key.annotated_decl`

The other (`setter_accessibility`, `nameoffset`, `kind` etc.) are present.

Hope this helps.

**Comment by Ross-Gibson** [comment]
I.m having a similar issue too with /Library/Ruby/Gems/2.0.0/gems/jazzy-0.1.2/lib/jazzy/sourcekitten.rb:41:in `+': no implicit conversion of nil into String (TypeError)

**Comment by juangamnik** [comment]
@Ross-Gibson same for me: `--skip-undocumented` works.

Hence, the `nil` value seems to come from an undocumented declaration. Please add an output, which declaration is responsible for that `nil` value. The respective information might be the breakthrough ;).

**Comment by juangamnik** [comment]
Perhaps another interesting information: with `--skip-undocumented` it does not throw any exceptions, but the generated docs just contain my `README.MD` and nothing else. As if I had no documentation at all.

**Comment by robb** [comment]
FWIW, I get the same error running `jazzy` on https://github.com/robb/Cartography/commit/7419c2a6d4072533f2b9d4de0eecd920c5ed57b1.

**Comment by dcharbonnier** [comment]
+1... sounds interesting but with this problem useless...

**Comment by jpsim** [comment]
I pushed a "fix" in #192. We'll be pushing a release to rubygems in the next few days. Until then, you can run from source.

Please let me know if you're still experiencing related issues.

**Comment by juangamnik** [comment]
Seems to be fixed for me (jazzy v0.2.0)
